src/Native/System.Native/pal_utilities.cpp

```diff
@@ -16,3 +18,13 @@ void SafeStringCopy(char* destination, int32_t destinationSize, char* source)
     snprintf(destination, destinationSize, "%s", source);
 #endif
 }
+
+extern "C"
+int32_t SnPrintF(char* string, int64_t size, char* format, ...)
```

Comment:
size should be unsigned as it's size_t underneath. Please check all of the int64_t's off_t is signed, but size_t is not.

We should add -Wconversion to our complilation. Apparently, it isn't included in -Wall. :( I'll do it in a separate change.